pkg/operator/ceph/object/status.go

```diff
@@ -71,6 +71,7 @@ func updateStatusBucket(client client.Client, name types.NamespacedName, phase c
 	}
 
 	objectStore.Status.BucketStatus = toCustomResourceStatus(objectStore.Status.BucketStatus, details, phase)
+	objectStore.Status.Phase = phase
```

Comment:
We have two types of status that we shouldn't mix.
1. Reconciling the object store
1. The bucket health check
If the object store successfully reconciles, then the phase should reflect it. If the bucket health check fails, it should just influence the bucket health check. Let's not change the phase based on bucket health.

Reply:
The problem is that the status phase after reconciling does not give us any guarantee and I believe some UIs are using it to determine the overall state...

Reply:
@anmolsachan can you confirm?

Reply:
Shouldn't we be using conditions instead of phase anyway? To avoid the overlap, the reconcile could use a different condition from the health check.

Reply:
UI takes the health information from the status field and shows on the dashboard. 
It is the ocs-operator that uses the phase information to mark the storage cluster ready. It is the OCS-operator that needs to know the basis on which it can mark the ceph cluster as connected and move on in the reconcile. The UI reflects the states of the OCS operator then.

Reply:
This really needs further discussions. Hopefully today.